From 5262ea57161c3527056385498721adc7f03db8c9 Mon Sep 17 00:00:00 2001
From: Szymon Mielczarek <szymonx.mielczarek@intel.com>
Date: Fri, 9 Aug 2019 14:17:21 +0200
Subject: [PATCH 1177/1214] ASoC: dapm: Allow creation of all mixer widget
 controls

Mixer widgets, like other widget types, may have extra controls
associated with them, e.g. a byte control used for configuration or
debugging purpose.
Thus far, creation of such controls was limited for mixer widgets,
i.e. control name had to be the same as mixer path name.

Let's create kcontrol objects for all associated controls.

Change-Id: I625f6e430ebe5f49c18d68d234ead02f7f499364
Tracked-On: OAM-84798
Signed-off-by: Szymon Mielczarek <szymonx.mielczarek@intel.com>
---
 sound/soc/soc-dapm.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index ba8298d..f48d872 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -977,18 +977,17 @@ static int dapm_new_mixer(struct snd_soc_dapm_widget *w)
 
 	/* add kcontrol */
 	for (i = 0; i < w->num_kcontrols; i++) {
+		if (!w->kcontrols[i]) {
+			ret = dapm_create_or_share_kcontrol(w, i);
+			if (ret < 0)
+				return ret;
+		}
 		/* match name */
 		snd_soc_dapm_widget_for_each_source_path(w, path) {
 			/* mixer/mux paths name must match control name */
 			if (path->name != (char *)w->kcontrol_news[i].name)
 				continue;
 
-			if (!w->kcontrols[i]) {
-				ret = dapm_create_or_share_kcontrol(w, i);
-				if (ret < 0)
-					return ret;
-			}
-
 			dapm_kcontrol_add_path(w->kcontrols[i], path);
 
 			data = snd_kcontrol_chip(w->kcontrols[i]);
-- 
2.7.4

